events/sheets-listener.js

```diff
@@ -0,0 +1,139 @@
+            const embedMessagesArray = Array.from(embedMessages.values());
+            console.log('Embed Messages Array:', embedMessagesArray);
+            
+            const sortedEmbeds = embedMessagesArray.sort((a, b) => b.createdTimestamp - a.createdTimestamp);
+            
+            const secondMostRecentEmbed = sortedEmbeds[1];
+
+            if (secondMostRecentEmbed) {
+              console.log(`Found second-to-last embed message in ${targetThread.name}`);
+              const auth = await google.auth.getClient({
+                keyFile: './credentials/baja-to-do-bot-fef8bb884c17.json',
+                scopes: 'https://www.googleapis.com/auth/spreadsheets',
+              });
+
+              console.log('Authenticated with Google Sheets API');
+
+              const sheets = google.sheets({ version: 'v4', auth });
+              const spreadsheetId = '1pb2W0BvAOMFeM4AXIbLzxM0dWJGYtqago8_8J4S5wEI';
+
+              try {
+                const sheetName = channelKeyword; 
+                console.log(`Processing sheet: ${sheetName}`);
+                const response = await sheets.spreadsheets.values.get({
+                  spreadsheetId,
+                  range: sheetName,
+                });
+
+                console.log(`Got response for sheet: ${sheetName}`);
+                // Get a list of the names of all the columns
+                const columns = sheetValuesResponse.data.values[0];
+            
+                const currentDate = new Date();
+                const [thisWeekStartDate, thisWeekEndDate, nextWeekStartDate, nextWeekEndDate] = calculateDates(currentDate);
+            
+                // Create an array of task objects, each task object is basically a row of the spreadsheet, keys are column names
+                const sheetRows = sheetData.map(row => {
+                    const taskObject = {};
+                    columns.forEach((columnName, index) => {
+                        taskObject[columnName] = row[index];
+                    });
+                    return taskObject;
+                });
+
+                const thisWeekTasks = sheetRows.filter(row => {
+                  const dueDate = new Date(row['Due Date']);
+                  return (dueDate >= thisWeekStartDate && dueDate <= thisWeekEndDate);
+              });
+
+                console.log(`Processed sheet data for: ${sheetName}`);
+                const updatedEmbed = formatTasks(thisWeekTasks, sheetName, sheetColor);
+                await secondMostRecentEmbed.edit({ embeds: [updatedEmbed] });
+
+                console.log(`Edited second-to-last embed for sheet "${sheetName}" in ${targetThread.name} thread.`);
+              } catch (error) {
+                console.error(`Error processing sheet ${sheetName}:`, error);
+              }
+            } else {
+              console.log(`No embeds found in the bot's last 100 messages in ${targetThread.name} thread.`);
```

Comment:
Just for general structure of code to understand / ease of reading it in the future, these if statements should probably check the negative, and then return early if trigerred. Then the rest of your code isn't indented in this big triple if statement stuff. Something like
```
func () => {
   if (errorHappens) {
       console.log("An error occured")
       return;
   }
   logic for your function...
}
```